self_hosted/ast.jou

```diff
@@ -133,10 +153,10 @@ class AstNameTypeValue:
 
 class AstSignature:
     funcname_location: Location
-    funcname: byte[100]
+    funcname: byte[100]  # TODO: rename to function_or_method_name
```

Comment:
Might as well do now

Reply:
Actually I'll make it `name` so that I don't have `function_or_method_name_location`